jena-db/jena-tdb2/src/main/java/org/apache/jena/tdb2/xloader/ProcBuildNodeTableX.java

```diff
@@ -100,26 +137,31 @@
         OutputStream toSortOutputStream;
         InputStream fromSortInputStream;
 
+        if ( sortThreads <= 0 )
+            sortThreads = 2;
+
         // ** Step 2: The sort
         Process proc2;
         try {
             //LOG.info("Step : external sort");
-            List<String> sortCmdBasics = Arrays.asList(
+            // Mutable list.
+            List<String> sortCmd = new ArrayList<>(Arrays.asList(
                   "sort",
-                    "--temporary-directory="+loaderFiles.TMPDIR, "--buffer-size=50%",
-                    "--parallel=2", "--unique",
-                    //"--compress-program=/usr/bin/gzip",
+                    "--temporary-directory="+loaderFiles.TMPDIR,
+                    "--buffer-size=50%",
+                    "--parallel="+sortThreads,
+                    "--unique",
                     "--key=1,1"
-                    );
+                    ));
 
-            List<String> sortCmd = new ArrayList<>(sortCmdBasics);
+            if ( BulkLoaderX.CompressSortNodeTableFiles )
+                sortCmd.add("--compress-program=/usr/bin/gzip");
```

Comment:
Huh, went to check that path... is it `/usr/bin/gzip` on mac? Or maybe depends on *nix flavour/distro? For me on Ubuntu 20.04 LTS:

```bash
kinow@ranma:~$ which gzip
/bin/gzip
kinow@ranma:~$ /usr/bin/gzip
bash: /usr/bin/gzip: No such file or directory
```

Reply:
On my machine `/bin` is a symbolic link to `/usr/bin`. Ubuntu 21.10. 

Here we are:
https://itectec.com/ubuntu/ubuntu-are-bin-and-sbin-now-symlinks-in-ubuntu-20-04/

New installations only at 20.04. On my machine upgrade seems to have done it at 21.04 judging by the timestamp.

At the moment, only Linux is supported. Nowadays, OSX sort(1) does have `--parallel` but I don't know how far back that goes. Given xloader is only for very large files (1B+) for many hour loads, 

I don't have access to a mac to test. All our testing has been on Linux servers except for my tests on a laptop.

"/usr/bin/gzip" is taken from the Linux man page for sort(1) but of course I read the 21.x man page.

I'll try to make it detech which.